api/v1/services/ProjectService.js

```diff
@@ -0,0 +1,208 @@
+
+const projectKeys = ["name", "description", "repo", "isPublished"];
+
+/**
+ * Checks to see if a requestor valid permissions to create a new project
+ * @param  {User} user creating the new project
+ * @return {Promise} resolving to true if the user is an organizer
+ * @throws InvalidParameterError when a user does not have correct permissions
+ */
+module.exports.canCreateProject = function (creator) {
+	if(creator.hasRole(roles.SUPERUSER) || creator.hasRole(roles.ORGANIZERS)){
+		return _Promise.resolve(true);
+	}
+
+	var message = "A project cannot be created with the provided credentials";
+	return _Promise.reject(new errors.UnauthorizedError(message));
+}
+
+
+/**
+ * Creates a project with the specificed attributes
+ * @param  {Object} Contains name, description, repo, and is_published
+ * @return {Promise} resolving to the newly-created project
+ * @throws InvalidParameterError when a project exists with the specified name
+ */
+module.exports.createProject = function (attributes) {
+	if(typeof attributes.repo === 'undefined'){
+		attributes.description = '';
+	}
+	if(typeof attributes.isPublished === 'undefined'){
+		attributes.isPublished = 0; //false
+	}
+
+	validationName = attributes.name.toLowerCase();
+
+	var project = Project.forge(attributes);
+	return project
+		.validate()
+		.catch(Checkit.Error, utils.errors.handleValidationError)
+		.then(function (validated) {
+			return Project.findByName(validationName);
+		})
+		.then(function (result){
+			if (!_.isNull(result)) {
+				var message = "A project with the given name already exists";
+				var source = "name";
+				throw new errors.InvalidParameterError(message, source);
+			}
+
+			return project
+				.save()
+				.then(function (project) {
+					return project;
+				});
+		})
+}
+
+/**
+ * Returns a project with the specified project id
+ * @param  {int} ID of the project
+ * @return {Promise} resolving to the project
+ * @throws InvalidParameterError when a project doesn't exist with the specified ID
+ */
+module.exports.findProjectById = function (id) {
+	return Project
+		.findById(id)
+		.then(function (result) {
+			if(_.isNull(result)){
+				var message = "A project with the given ID cannot be found";
+				var source = "id";
+				throw new errors.NotFoundError(message, source);
+			}
+
+			return _Promise.resolve(result);
+		});
+}
+
+/**
+ * Update a key value pair in a project
+ * @param  {Project} Project that will be updated
+ * @param  {String} Key is the name of the attribute
+ * @param  {String} Value is the new value of the attribute
+ * @return {Promise} resolving to the updated project
+ * @throws InvalidParameterError when the key is not valid
+ */
+module.exports.updateProject = function (project, key, value) {
+	if(projectKeys.indexOf(key.toLowerCase()) == -1){
+		var message = "The given key is invalid";
+		var source = "key";
+		throw new errors.InvalidParameterError(message, source);
+	}
+	return project
+		.set(key, value)
```

Comment:
When we do puts (updates), we want to pass up the entire updated object. So you should be expecting to set multiple key-value pairs, and design your function like that. Take a look at the `Mentor` PUT code/documentation for a full example.